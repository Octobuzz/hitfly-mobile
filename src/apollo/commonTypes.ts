import { Track } from './schemas'

export type HeaderMode = 'light' | 'dark'

export type NavigationState = 'main' | 'profile' | 'auth'

export interface HeaderSettings {
  mode: HeaderMode
  state: NavigationState
}

export type NullableTrack = Track | null
