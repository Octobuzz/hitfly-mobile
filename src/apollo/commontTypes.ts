export type CollectionsType = 'recommended' | 'musicFan'
