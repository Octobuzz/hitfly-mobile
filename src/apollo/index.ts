import * as _storage from './storage'
export const storage = _storage

export * from './queries'
export * from './schemas'
export * from './mutations'
export * from './commonTypes'

export { default as setupClient } from './setupClient'
