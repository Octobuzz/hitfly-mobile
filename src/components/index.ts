export * from './views'
export * from './fields'
export * from './buttons'

export { default as TextBase } from './TextBase'
export { default as GenreItem } from './GenreItem'
export { default as TextWithLines } from './TextWithLines'
