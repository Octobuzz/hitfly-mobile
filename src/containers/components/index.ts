export * from './SocialAuth'

export { default as SocialAuth } from './SocialAuth'
export { default as HeaderRightButtons } from './HeaderRightButtons'
