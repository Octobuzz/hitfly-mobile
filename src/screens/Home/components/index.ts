export { default as StarsSection } from './StarsSection'
export { default as TracksSection } from './TracksSection'
export { default as SectionHeader } from './SectionHeader'
export { default as SectionWrapper } from './SectionWrapper'
export { default as PlaylistHeader } from './PlaylistHeader'
export { default as CollectionSection } from './CollectionSection'
