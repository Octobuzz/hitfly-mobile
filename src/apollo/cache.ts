import { InMemoryCache } from 'apollo-cache-inmemory'
import { persistCache } from 'apollo-cache-persist'
import { storageInstance } from 'src/utils'
import gql from 'graphql-tag'

export const defaults = {
  headerSettings: {
    mode: 'dark',
    state: 'main',
    __typename: 'HeaderSettings',
  },
  isPlaying: false,
  activeTrack: null,
  playlist: [],
}

export const typeDefs = gql`
  type Track {
    id: Int!
    title: String
    singer: String
    date: String
    songText: String
    fileUrl: String
    musicWave: [Int]
    length: Int
  }
  type HeaderSettings {
    style: String!
    state: String!
  }
`

export default async (): Promise<InMemoryCache> => {
  const cache = new InMemoryCache({
    freezeResults: true,
  })

  await persistCache({
    cache,
    debug: __DEV__,
    // @ts-ignore
    storage: storageInstance,
  })

  return cache
}
