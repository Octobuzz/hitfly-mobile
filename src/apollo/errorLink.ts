import { Alert } from 'react-native'
import { Operation } from 'apollo-link'
import { GraphQLError } from 'graphql'
import { onError } from 'apollo-link-error'
import { AuthErrorModalRef } from 'src/globalRefs'

const handle401Error = (error: GraphQLError, operation: Operation) => {
  // FIXME: это костыль так как я не понял почему не вытаскивает профиль
  // пришедший после мутации логина
  if (operation.operationName === 'ProfileAvatar') {
    return
  }
  AuthErrorModalRef.show()
}

const handleGraphQLErrors = (
  errors: readonly GraphQLError[],
  operation: Operation,
) => {
  errors.forEach(error => {
    const { message } = error
    // костыль с бэка, по другому ошибку не выявить
    if (message === 'Unauthorized') {
      handle401Error(error, operation)
    }
  })
}

export default onError(({ networkError, graphQLErrors, operation }) => {
  if (networkError) {
    Alert.alert('Ошибка сети', networkError.message)
  }

  if (graphQLErrors) {
    handleGraphQLErrors(graphQLErrors, operation)
  }
})
