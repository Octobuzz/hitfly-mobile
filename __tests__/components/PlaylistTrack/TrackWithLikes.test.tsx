import React from 'react'
import {
  render,
  fireEvent,
  ReactTestInstanceExtended,
} from '../../../jest/test-utils'
import { TrackWithLikes } from 'src/components'
import { Track } from 'src/apollo'

describe('TrackWithLikes', () => {
  const track = {
    id: 1,
    cover: [
      {
        imageUrl: 'some url',
      },
    ],
    title: 'Some track title',
    group: {
      title: 'Some group title',
    },
  } as Track

  it('renders correctly', () => {
    const { container } = render(<TrackWithLikes track={track} />)
    expect(container.children[0] as ReactTestInstanceExtended).toMatchSnapshot()
  })

  it('renders correctly in playing state', () => {
    const { container } = render(<TrackWithLikes isPlaying track={track} />)
    expect(container.children[0] as ReactTestInstanceExtended).toMatchSnapshot()
  })

  it('renders correctly when liked', () => {
    const likedTrack = { ...track, isFavourite: true } as Track
    const { container } = render(<TrackWithLikes track={likedTrack} />)
    expect(container.children[0] as ReactTestInstanceExtended).toMatchSnapshot()
  })

  it('can be pressed', () => {
    const onPress = jest.fn()
    const { getByTestId } = render(
      <TrackWithLikes onPress={onPress} track={track} />,
    )
    fireEvent.press(getByTestId('wrapper'))
    expect(onPress).toBeCalled()
  })

  it('can be pressed in playing state', () => {
    const onPress = jest.fn()
    const { getByTestId } = render(
      <TrackWithLikes isPlaying onPress={onPress} track={track} />,
    )
    fireEvent.press(getByTestId('wrapper'))
    expect(onPress).toBeCalled()
  })

  it('can be liked', () => {
    const onPressLike = jest.fn()
    const { getByTestId } = render(
      <TrackWithLikes onPressLike={onPressLike} track={track} />,
    )
    fireEvent.press(getByTestId('like'))
    expect(onPressLike).toBeCalled()
  })

  it('can be unliked', () => {
    const onPressLike = jest.fn()
    const likedTrack = { ...track, isFavorite: true } as Track
    const { getByTestId } = render(
      <TrackWithLikes onPressLike={onPressLike} track={likedTrack} />,
    )
    fireEvent.press(getByTestId('like'))
    expect(onPressLike).toBeCalled()
  })
})
