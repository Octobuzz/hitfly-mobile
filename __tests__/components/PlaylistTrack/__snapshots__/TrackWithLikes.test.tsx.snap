// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TrackWithLikes renders correctly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingBottom": 12,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 12,
      },
    ]
  }
  testID="wrapper"
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "borderRadius": 4,
              "height": 32,
              "width": 32,
            },
          ],
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "uri": "some url",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginHorizontal": 16,
        },
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "GothamPro-Bold",
            "fontSize": 12,
            "lineHeight": 18,
          },
        ]
      }
    >
      Some track title
    </Text>
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "GothamPro",
            "fontSize": 10,
            "lineHeight": 12,
          },
        ]
      }
    />
  </View>
  <TouchableOpacity
    activeOpacity={0.8}
    hitSlop={
      Object {
        "bottom": 10,
        "left": 10,
        "right": 10,
        "top": 10,
      }
    }
    style={
      Array [
        Object {},
      ]
    }
    testID="like"
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/images/icons/heart.png",
        }
      }
      style={
        Array [
          Object {
            "tintColor": "#ffffff",
          },
        ]
      }
    />
  </TouchableOpacity>
</TouchableOpacity>
`;

exports[`TrackWithLikes renders correctly in playing state 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingBottom": 12,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 12,
      },
    ]
  }
  testID="wrapper"
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "borderRadius": 4,
              "height": 32,
              "width": 32,
            },
          ],
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "uri": "some url",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "rgba(49, 49, 49, 0.7)",
            "borderRadius": 4,
            "bottom": 0,
            "justifyContent": "center",
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          },
        ]
      }
    >
      <Icon
        color="#ffffff"
        name="pause"
        size={16}
        style={
          Array [
            Object {},
          ]
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginHorizontal": 16,
        },
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "GothamPro-Bold",
            "fontSize": 12,
            "lineHeight": 18,
          },
        ]
      }
    >
      Some track title
    </Text>
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "GothamPro",
            "fontSize": 10,
            "lineHeight": 12,
          },
        ]
      }
    />
  </View>
  <TouchableOpacity
    activeOpacity={0.8}
    hitSlop={
      Object {
        "bottom": 10,
        "left": 10,
        "right": 10,
        "top": 10,
      }
    }
    style={
      Array [
        Object {},
      ]
    }
    testID="like"
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/images/icons/heart.png",
        }
      }
      style={
        Array [
          Object {
            "tintColor": "#ffffff",
          },
        ]
      }
    />
  </TouchableOpacity>
</TouchableOpacity>
`;

exports[`TrackWithLikes renders correctly when liked 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  style={
    Array [
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "paddingBottom": 12,
        "paddingLeft": 16,
        "paddingRight": 16,
        "paddingTop": 12,
      },
    ]
  }
  testID="wrapper"
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Array [
            Object {
              "borderRadius": 4,
              "height": 32,
              "width": 32,
            },
          ],
        ]
      }
    >
      <FastImageView
        resizeMode="cover"
        source={
          Object {
            "uri": "some url",
          }
        }
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      />
    </View>
  </View>
  <View
    style={
      Array [
        Object {
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
          "marginHorizontal": 16,
        },
      ]
    }
  >
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "GothamPro-Bold",
            "fontSize": 12,
            "lineHeight": 18,
          },
        ]
      }
    >
      Some track title
    </Text>
    <Text
      numberOfLines={1}
      style={
        Array [
          Object {
            "color": "#ffffff",
            "fontFamily": "GothamPro",
            "fontSize": 10,
            "lineHeight": 12,
          },
        ]
      }
    />
  </View>
  <TouchableOpacity
    activeOpacity={0.8}
    hitSlop={
      Object {
        "bottom": 10,
        "left": 10,
        "right": 10,
        "top": 10,
      }
    }
    style={
      Array [
        Object {},
      ]
    }
    testID="like"
  >
    <Image
      resizeMode="contain"
      source={
        Object {
          "testUri": "../../../src/assets/images/icons/heart.png",
        }
      }
      style={
        Array [
          Object {
            "tintColor": "#ffffff",
          },
        ]
      }
    />
  </TouchableOpacity>
</TouchableOpacity>
`;
